Src/Application/CaseStudy.API/Controllers/HeaderFooterSettingsController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using CaseStudy.Core.Contracts.IUnitOfWork;
+using CaseStudy.Core.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CaseStudy.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class HeaderFooterSettingsController : ControllerBase
+    {
+        private IHeaderFooterSettingsServices services;
+        public HeaderFooterSettingsController(IHeaderFooterSettingsServices headerFooterSettingsServices)
+        {
+            services = headerFooterSettingsServices;
+        }
+        [HttpGet]
+        public async Task<IActionResult> GetHeaderFoooterSettings()
+        {
+            var settings =  await services.GetHeaderFooterSettings();
+            if (settings == null)
+            {
+                return NotFound("No records Found");
+            }
+            return Ok(settings);
+        }
+        [HttpGet("/{DealerId}")]
+        public async Task<IActionResult> GetHeaderFoooterSettingsById( int DealerId)
+        {
+            var settings = await services.GetHeaderFooterSettingsById(DealerId);
+            if (settings == null)
+            {
+                return NotFound("No records Found");
+            }
+            return Ok(settings);
+        }
+        [HttpPost]
+        public async Task<IActionResult> AddHeaderFooterSettings(HeaderAndFooterSettings headerAndFooterSettings)
+        {
+            var settings = await services.AddHeaderAndFooterSettings(headerAndFooterSettings);
+            if (!settings)
+            {
+                return BadRequest("Dealer Not Found, Cannot add the record");
+            }
+            return Ok(headerAndFooterSettings);
+        }
+        [HttpPut]
```

Comment:
HttpPut -> to update a record you should take id as input 
![image](https://github.com/SSowmya7/CaseStudy/assets/113988510/03c4501d-4cdf-4a2d-8e2f-5dab78bebbb2)
https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design, https://code-maze.com/aspnetcore-webapi-best-practices/

Reply:
okay ,Sure.